[CI] SearchTemplateRequestTests testConcurrentSerialization failing

**Build Scans:**
- [elasticsearch-periodic #5762 / release-tests](https://gradle-enterprise.elastic.co/s/7csqav5yweudi)
- [elasticsearch-pull-request #49522 / release-tests](https://gradle-enterprise.elastic.co/s/mvsh4qk5gyhpw)
- [elasticsearch-periodic #5757 / release-tests](https://gradle-enterprise.elastic.co/s/scvvaopiswqhu)

**Reproduction Line:**
```
./gradlew ":modules:lang-mustache:test" --tests "org.elasticsearch.script.mustache.SearchTemplateRequestTests.testConcurrentSerialization" -Dtests.seed=2A2139F411B3F660 -Dbuild.snapshot=false -Dtests.jvm.argline="-Dbuild.snapshot=false" -Dlicense.key=x-pack/license-tools/src/test/resources/public.key -Dtests.locale=ff-Adlm-GN -Dtests.timezone=Pacific/Niue -Druntime.java=23
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.script.mustache.SearchTemplateRequestTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(testConcurrentSerialization),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: <org.elasticsearch.script.mustache.SearchTemplateRequest/unset>
     but: was <org.elasticsearch.script.mustache.SearchTemplateRequest/unset>
```

**Issue Reasons:**
- [main] 3 consecutive failures in step release-tests
- [main] 3 failures in test testConcurrentSerialization (1.2% fail rate in 256 executions)
- [main] 3 failures in step release-tests (37.5% fail rate in 8 executions)
- [main] 2 failures in pipeline elasticsearch-periodic (50.0% fail rate in 4 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -248,30 +248,6 @@ tests:
 - class: org.elasticsearch.xpack.esql.optimizer.LocalPhysicalPlanOptimizerTests
   method: testSingleMatchOperatorFilterPushdownWithStringValues {default}
   issue: https://github.com/elastic/elasticsearch/issues/119721
-- class: org.elasticsearch.script.mustache.SearchTemplateRequestTests
-  method: testConcurrentSerialization
-  issue: https://github.com/elastic/elasticsearch/issues/119819
-- class: org.elasticsearch.script.mustache.SearchTemplateRequestTests
-  method: testEqualsAndHashcode
-  issue: https://github.com/elastic/elasticsearch/issues/119820
-- class: org.elasticsearch.script.mustache.SearchTemplateRequestTests
-  method: testConcurrentEquals
-  issue: https://github.com/elastic/elasticsearch/issues/119821
-- class: org.elasticsearch.script.mustache.SearchTemplateRequestTests
-  method: testSerialization
-  issue: https://github.com/elastic/elasticsearch/issues/119822
-- class: org.elasticsearch.index.rankeval.RankEvalRequestTests
-  method: testSerialization
-  issue: https://github.com/elastic/elasticsearch/issues/119859
-- class: org.elasticsearch.index.rankeval.RankEvalRequestTests
-  method: testEqualsAndHashcode
-  issue: https://github.com/elastic/elasticsearch/issues/119860
-- class: org.elasticsearch.index.rankeval.RankEvalRequestTests
-  method: testConcurrentSerialization
-  issue: https://github.com/elastic/elasticsearch/issues/119861
-- class: org.elasticsearch.index.rankeval.RankEvalRequestTests
-  method: testConcurrentEquals
-  issue: https://github.com/elastic/elasticsearch/issues/119862
 
 # Examples:
 #

--- a/server/src/main/java/org/elasticsearch/action/support/IndicesOptions.java
+++ b/server/src/main/java/org/elasticsearch/action/support/IndicesOptions.java
@@ -884,7 +884,8 @@ public record IndicesOptions(
         if (ignoreUnavailable()) {
             backwardsCompatibleOptions.add(Option.ALLOW_UNAVAILABLE_CONCRETE_TARGETS);
         }
-        if (allowSelectors()) {
+        // Until the feature flag is removed we access the field directly from the gatekeeper options.
+        if (gatekeeperOptions().allowSelectors()) {
             if (out.getTransportVersion()
                 .between(TransportVersions.V_8_14_0, TransportVersions.REPLACE_FAILURE_STORE_OPTIONS_WITH_SELECTOR_SYNTAX)) {
                 backwardsCompatibleOptions.add(Option.ALLOW_FAILURE_INDICES);
@@ -1464,7 +1465,8 @@ public record IndicesOptions(
             + ignoreAliases()
             + ", ignore_throttled="
             + ignoreThrottled()
-            + (DataStream.isFailureStoreFeatureFlagEnabled() ? ", allow_selectors=" + allowSelectors() : "")
+            // Until the feature flag is removed we access the field directly from the gatekeeper options.
+            + (DataStream.isFailureStoreFeatureFlagEnabled() ? ", allow_selectors=" + gatekeeperOptions().allowSelectors() : "")
             + ']';
     }
 }

